From 6a3ca0144bdef940499d458b42c7907c20f69bf2 Mon Sep 17 00:00:00 2001
From: shengfei Xu <xsf@rock-chips.com>
Date: Sun, 8 Nov 2020 17:37:14 +0800
Subject: [PATCH] arm64: dts: rockchip: fix init voltage for rk3568-evb

Add regulator-init-microvolt for driver to init the regulator

Signed-off-by: shengfei Xu <xsf@rock-chips.com>
Change-Id: Ie967a0cccf63b3ff6966caedecdc58d469f08d2d
---
 arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi | 23 +++++++++++++-------
 1 file changed, 15 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi
index 52b278e928ee..d2021745cd87 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi
@@ -725,8 +725,9 @@
 			vdd_logic: DCDC_REG1 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <900000>;
+				regulator-min-microvolt = <500000>;
 				regulator-max-microvolt = <1350000>;
+				regulator-init-microvolt = <900000>;
 				regulator-ramp-delay = <6001>;
 				regulator-initial-mode = <0x2>;
 				regulator-name = "vdd_logic";
@@ -739,8 +740,9 @@
 			vdd_gpu: DCDC_REG2 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <900000>;
-				regulator-max-microvolt = <900000>;
+				regulator-min-microvolt = <500000>;
+				regulator-max-microvolt = <1350000>;
+				regulator-init-microvolt = <900000>;
 				regulator-ramp-delay = <6001>;
 				regulator-initial-mode = <0x2>;
 				regulator-name = "vdd_gpu";
@@ -759,20 +761,25 @@
 				};
 			};
 
-			vcc_npu: DCDC_REG4 {
-				regulator-min-microvolt = <900000>;
-				regulator-max-microvolt = <900000>;
+			vdd_npu: DCDC_REG4 {
+				regulator-always-on;
+				regulator-boot-on;
+				regulator-min-microvolt = <500000>;
+				regulator-max-microvolt = <1350000>;
+				regulator-init-microvolt = <900000>;
+				regulator-ramp-delay = <6001>;
 				regulator-initial-mode = <0x2>;
-				regulator-name = "vcc_npu";
+				regulator-name = "vdd_npu";
 				regulator-state-mem {
 					regulator-off-in-suspend;
 				};
 			};
 
 			vdda0v9_image: LDO_REG1 {
+				regulator-boot-on;
+				regulator-always-on;
 				regulator-min-microvolt = <900000>;
 				regulator-max-microvolt = <900000>;
-				regulator-initial-mode = <0x1>;
 				regulator-name = "vdda0v9_image";
 				regulator-state-mem {
 					regulator-off-in-suspend;
-- 
2.35.3

